vowpalwabbit/learner.h

```diff
@@ -311,7 +441,10 @@ typedef void (*tend_example)(vw& all, void* d, example& ec);
  template<class T> learner<T>& init_learner(base_learner* base,
                                             void(*learn)(T&, base_learner&, example&),
                                             void(*predict)(T&, base_learner&, example&))
-   { return init_learner<T>(nullptr, base, learn, predict, 1, base->pred_type); }
+ {
+   typedef void(*fptr)(T&, base_learner&, multi_ex&);  // compile hint to resolve the right template
```

Comment:
It seems like we should avoid requiring a multi_ex function pointer to eliminate all these repeated typedefs?